onnxruntime/core/optimizer/insert_cast_transformer.cc

```diff
@@ -237,17 +237,34 @@ Status InsertCastTransformer::ApplyImpl(onnxruntime::Graph& graph, bool& modifie
       }
     }
 
-    if (casted && node->GetExecutionProviderType().empty()) {
-      //set current node to CPU execution provider
+    if (casted) {
```

Comment:
removed the redundant check `node->GetExecutionProviderType().empty()`. This is already ensured in the `NeedInsertCast()` which has to be true for a node to be even a candidate for the InsertCastTransformer.